tests/cpp/models/qwen3_embedding/test_qwen3_embedding_pipeline.cpp

```
@@ -0,0 +1,75 @@
+}
+
+void test_kernel_filename_component_cannot_escape_temp_directory() {
+    const auto value = trtmc::sanitize_kernel_filename_component("../flashinfer/decode.so");
+    if (value.find('/') != std::string::npos || value.find("..") != std::string::npos)
+        throw std::runtime_error("kernel filename sanitizer preserved traversal syntax");
+}
+
+} // namespace
+
+int main() {
+    test_last_token_pool_handles_right_padding();
```

Cppcheck (2.21.0)

[error] Unhandled exception thrown in function that is an entry point.

(throwInEntryPoint)